app/models/network/conserving_producer.rb

```diff
@@ -5,18 +5,27 @@ class ConservingProducer < Technology
     extend Disableable
 
     def self.disabled?(options)
-      ! options[:flexibility]
+      options[:capping_solar_pv] == 'false'
     end
 
     def self.disabled_class
       Technology
     end
 
-    def conditional_consumption_at(frame)
-      production_at(frame)
+    def production_at(frame)
+      production = super
+      capping = (capacity * capping_percentage).abs
+
+      production > capping ? capping : production
     end
 
     def store(_frame, _amount)
     end
+
+    private
+
+      def capping_percentage
+        options['capping_solar_pv_slider_value'].to_f
```

Comment:
Same here as above with `"false"`. The Rails-side should be casting these values before creating the Network (in fact, probably before saving them to the database).